_inc/lib/core-api/wpcom-endpoints/service-api-keys.php

```diff
@@ -109,18 +112,31 @@ public function get_public_item_schema() {
 	 * }
 	 */
 	public static function get_service_api_key( $request ) {
-
 		$service = self::validate_service_api_service( $request['service'] );
 		if ( ! $service ) {
 			return self::service_api_invalid_service_response();
 		}
-		$option  = self::key_for_api_service( $service );
+
+		switch ( $service ) {
+			case 'mapbox':
+				$mapbox                 = self::get_service_api_key_mapbox();
+				$service_api_key        = $mapbox['key'];
+				$service_api_key_source = $mapbox['source'];
+				break;
+			default:
+				$option                 = self::key_for_api_service( $service );
+				$service_api_key        = Jetpack_Options::get_option( $option, '' );
+				$service_api_key_source = '';
```

Comment:
Since we set `source` to `site` elsewhere, should we default to `site` here as well?